### PR TITLE
Sdl must process system request for request type climate

### LINKED ISSUE #2426: SDL must processing SystemRequest for RequestType: CLIMATE with fileName:(binary file)
### Bug Report
SDL must processing SystemRequest for RequestType: CLIMATE with fileName:(binary file)

##### Reproduction Steps
1. Start SDL and HMI
2. Connect mobile and create mobile session
3. Register mobile app and activated
4. Mobile app send SystemRequest( RequestType: CLIMATE, fileName:(binary file)

##### Expected Behavior
SDL must processing SystemRequest for RequestType: CLIMATE, fileName:(binary file) with resultCode:"SUCCESS" on the mobile app

##### Observed Behavior
SDL respond with resultCode:"REJECTED" to mobile app

##### OS & Version Information
* OS/Version:
* SDL Core Version:

### PR DESCRIPTION
**Pull request by ZhdanovP**
Fixes #2426 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
- atf script is provided in PR [#2017](https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2017)

### Summary
According to requirements:
       "If the requestType = PROPRIETARY, add to mobile API fileType = JSON
        If the requestType = HTTP, add to mobile API fileType = BINARY"
Also in SDL we don't save the PT to file - we put it directly in binary_data - this functionality was provided in current pull request.

### Tasks Remaining:
- [x] implementation
- [x] provide unit tests
- [x] provide atf script

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc [hunk 1]
```diff
@@ -106,16 +106,16 @@ void OnSystemRequestNotification::Run() {
     }
   }
 
+  const std::string filename =
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc]: @ZhdanovP propose `file_name`

Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc]: fixed 19f25ae03d7ade886a15826d883f0b446ed71587

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc [hunk 2]
```diff
@@ -106,17 +106,17 @@ void OnSystemRequestNotification::Run() {
...
        "If the requestType = PROPRIETARY, add to mobile API fileType = JSON
         If the requestType = HTTP, add to mobile API fileType = BINARY"
        Also in Genivi SDL we don't save the PT to file - we put it directly in
        binary_data */
-
-    const std::string filename =
-        (*message_)[strings::msg_params][strings::file_name].asString();
-    BinaryMessage binary_data;
-    file_system::ReadBinaryFile(filename, binary_data);
+    LOG4CXX_DEBUG(logger_, "According to requirements");
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc]: @ZhdanovP Unclear log...Remove it please

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc [hunk 3]
```diff
@@ -132,6 +132,8 @@ void OnSystemRequestNotification::Run() {
       (*message_)[strings::msg_params][strings::timeout] =
           policy_handler.TimeoutExchangeSec();
     }
+  } else if (file_system::ReadBinaryFile(file_name, binary_data) == true) {
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc]: @ZhdanovP `if (bool func())`. Why do you use bool func() == true ?

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc [hunk 4]
```diff
@@ -106,17 +106,17 @@ void OnSystemRequestNotification::Run() {
...
         If the requestType = HTTP, add to mobile API fileType = BINARY"
        Also in Genivi SDL we don't save the PT to file - we put it directly in
        binary_data */
-
-    const std::string filename =
-        (*message_)[strings::msg_params][strings::file_name].asString();
-    BinaryMessage binary_data;
-    file_system::ReadBinaryFile(filename, binary_data);
+    LOG4CXX_DEBUG(logger_, "According to requirements");
+    file_system::ReadBinaryFile(file_name, binary_data);
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc]: @ZhdanovP Why don't you check return value here?

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc [hunk 5]
```diff
@@ -229,6 +229,125 @@ TEST_F(SystemRequestTest, Run_RequestTypeDisallowed_SendDisallowedResponse) {
   command->Run();
 }
 
+TEST_F(SystemRequestTest, Run_FileWithBinaryDataSaveUnsuccessful_GenericError) {
+  std::vector<uint8_t> binary_data;
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc]: @ZhdanovP Don't we have typedef fot this type if data?

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc [hunk 6]
```diff
@@ -229,6 +229,125 @@ TEST_F(SystemRequestTest, Run_RequestTypeDisallowed_SendDisallowedResponse) {
...
+  (*msg)[am::strings::msg_params][am::strings::file_name] = file_name;
+  (*msg)[am::strings::params][am::strings::binary_data] = binary_data;
+
+  PreConditions();
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId, request_type))
+      .WillOnce(Return(true));
+
+  const std::string kBinaryFolder = "test_binary_folder";
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc]: @ZhdanovP 
- named wrongly according to SDL Coding Style

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc [hunk 7]
```diff
@@ -229,6 +229,125 @@ TEST_F(SystemRequestTest, Run_RequestTypeDisallowed_SendDisallowedResponse) {
...
+  EXPECT_CALL(app_mngr_settings_, system_files_path())
+      .WillRepeatedly(ReturnRef(kBinaryFolder));
+  EXPECT_CALL(app_mngr_, SaveBinary(binary_data, kBinaryFolder, file_name, 0))
+      .WillOnce(Return((mobile_apis::Result::INVALID_DATA)));
+
+  const std::string info = "";
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc]: @ZhdanovP const std::string info;

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc [hunk 8]
```diff
@@ -229,6 +229,125 @@ TEST_F(SystemRequestTest, Run_RequestTypeDisallowed_SendDisallowedResponse) {
...
+          _));
+
+  std::shared_ptr<SystemRequest> command(CreateCommand<SystemRequest>(msg));
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(SystemRequestTest,
+       Run_FileWithoutBinaryDataNotHTTPTypeNotSystemFile_Rejected) {
+  const std::string kStorageFolder2 = "test_storage_folder2";
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc]: @ZhdanovP 
- named wrongly according to SDL Coding Style

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc (hunk 2 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc]: @AByzhynar please take a look f2dd4a6562711fbaaf71ad9b7cb53d9ea4c81c0e

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc (hunk 3 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc]: @AByzhynar please take a look f2dd4a6562711fbaaf71ad9b7cb53d9ea4c81c0e

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc (hunk 4 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_system_request_notification.cc]: @AByzhynar please take a look f2dd4a6562711fbaaf71ad9b7cb53d9ea4c81c0e

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc (hunk 8 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc]: @AByzhynar please take a look 8ed4a05b95d7bfc885e2bad85a6970636ef2e7f8

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc (hunk 7 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc]: @AByzhynar please take a look 8ed4a05b95d7bfc885e2bad85a6970636ef2e7f8

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc (hunk 6 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc]: @AByzhynar please take a look 8ed4a05b95d7bfc885e2bad85a6970636ef2e7f8

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc (hunk 5 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc]: @AByzhynar please take a look 8ed4a05b95d7bfc885e2bad85a6970636ef2e7f8

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc [hunk 9]
```diff
@@ -229,6 +229,125 @@ TEST_F(SystemRequestTest, Run_RequestTypeDisallowed_SendDisallowedResponse) {
...
+  MessageSharedPtr msg = CreateIVSUMessage();
+  const mobile_apis::RequestType::eType request_type =
+      mobile_apis::RequestType::CLIMATE;
+  (*msg)[am::strings::msg_params][am::strings::request_type] = request_type;
+  (*msg)[am::strings::msg_params][am::strings::file_name] = file_name;
+  (*msg)[am::strings::params][am::strings::binary_data] = binary_data;
+
+  PreConditions();
+
+  EXPECT_CALL(mock_policy_handler_,
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/system_request_test.cc]: @ZhdanovP Look like it is ON_CALL for this test

**Comment by AByzhynar** [comment]
@ZhdanovP Please check the reason of checks failure

**Comment by ZhdanovP** [comment]
@AByzhynar Checks were fixed.

**Comment by AByzhynar** [comment]
@ZhdanovP I can't see that all checks were fixed

**Comment by ShobhitAd** [comment]
Unable to reproduce the issue on the latest develop https://github.com/smartdevicelink/sdl_core/commit/0736ea16ce07caaf9ea83bfadb3b8e3eee950b3b. SDL correctly forwards the request to the HMI and returns a SUCCESS response

**Pull request closed without merging**